pkg/api/types.go

```diff
@@ -86,6 +86,18 @@ func (c ResourceConfiguration) RequirementsForStep(name string) ResourceRequirem
 	return req
 }
 
+func (c ResourceConfiguration) RequirementsForTemplateStep(name string) ResourceRequirements {
```

Comment:
I don't think this is correct -- instead of never using the default, we should only use it when the template itself does not set resource requirements